### ISSUE TITLE
Clarify http/https vocabulary differences with JSON-LD

### ISSUE THREAD

**Issue by stain** [schemaorg/schemaorg#2886]
https://schema.org/docs/developers.html seems to indicate that developers should prefer `https` version of the vocabulary definitions.

However the [JSON-LD context](https://schema.org/docs/jsonldcontext.json) (even with `@context: https://schema.org/`) maps to the `http` namespace which has been consistent since the beginning of schema.org (I do not suggest changing this!) - try for instance in https://json-ld.org/playground/ 

I think it's good the JSON-LD is consistent for both `@context` examples.

However RDFa examples like https://schema.org/CreativeWork use `vocab="https://schema.org/"` as namespace declaration.

I think `developers.html` should acknowledge this difference so that developers are aware of both variants, and highlight that the JSON-LD context will map to `http`. It depends on where the data comes from which variant developers will want.

Something like:

> For consistency with structured data already deployed with JSON-LD, the `@context` [resolved](https://schema.org/docs/jsonldcontext.json) for both `http://schema.org/` and `https://schema.org/`  will continue to be mapped to terms in the `http` namespace.

BTW: `developers.html` page could also benefit with a link to the JSON-LD context <https://schema.org/docs/jsonldcontext.json> as I struggled to find it!

**Comment by danbri** [comment]
You are right - we've also apparently lost navigability to the FAQ document.

The https thing is tricky, and needs documenting somewhere. I'll tag this issue for the next release - thanks

**Comment by mdw00d** [comment]
Changing the namespace is problematic for existing applications.  Obviously if a program was written referencing e.g., `schema:Action` and the definition of `schema:` has changed from `http://schema.org/` to `https://schema.org/`, then this adds unnecessary complexity for programs that now have to know that those two predicates are the same.   SPARQL queries have to handle either form, etc.

Namespaces really should not change unless there's a fundamentally new version.

**Comment by nicholascar** [comment]
rdflib will stick to HTTPS for schema.org. We had some confusion with an accidental reversion to HTTP recently but we are now HTTPS and will stay there.

**Comment by rlzijdeman** [comment]
In #3270:

> [@danbri](https://github.com/danbri) There is no in-built configuration in our cloud hosted site to serve a different static file dependant on the protocol (http / https) used for a request.
> 
> This leaves two options:
> 
> 1. Implement live logic to inspect incoming requests and return protocol specific context files. However, this would expose us to the type of DoS attacks that caused us to move to a static site.
> 2. Change the current context file to be https based which would satisfy the desires expressed here. However, the knock-on ramifications of such a move for those currently expecting and/or depending upon http based code could be significant.

So `1.` does not seem to be an option, but today's course of not acting on `2.` has ramifications of its own. In my country two major national research infrastructures are now split over the http:// - https:// issue (see https://github.com/codemeta/codemeta/issues/373), but also the many references above.

To sum up:
- the preferred URI for schema terms is https:// as mentioned [here](https://schema.org/docs/faq.html#19) and acknowledged [here](https://github.com/schemaorg/schemaorg/pull/2814) and [here](https://github.com/schemaorg/schemaorg/pull/2814)
- the schema documentation provides both a https:// and http:// json-ld version, the https:// being the default select option. See: [docs/developers]https://schema.org/docs/developers.html
- only known to insiders (as not reachable from the FAQ) there is also a [documentation file](https://schema.org/docs/jsonldcontext.jsonld) referring to the deprecated URI http:// , which is used as argument to stick to http:// when using JSON-LD  see [here](https://github.com/codemeta/codemeta/issues/373) and leads to various issues (e.g. [here](https://github.com/schemaorg/schemaorg/issues/3270)). 

I would therefore, like many above, reiterate the request to -take action- on this matter. Note, that this is perhaps more about social engineering than about technical engineering. [Schema.org](http://schema.org/) now ‘deprecating the http:// URI, but leaving ample room for discussion in the FAQ and ‘hidden’ doc, creates unclarity. 

A solution as with Python migrating from 2 to 3 seems a good option, using a clear announcement of the https:// being the official URI (also in JSON-LD) say per Jan 1st 2027, with supported being dropped by say Jan 1st 2028, gives: (1) clarity on the future direction of Schema, (2) ample time for people to adjust to the idea and make decisions on URI use in current and future projects, (3) brings the community 1 step closer to aligned vocabulary use.

**Comment by progval** [comment]
> only known to insiders (as not reachable from the FAQ) there is also a [documentation file](https://schema.org/docs/jsonldcontext.jsonld) referring to the deprecated URI http://

It's not a documentation file, it is machine-readable and used when someone writes `"@context": "https://schema.org"`:

```
$ curl -s https://schema.org/ -H 'Accept: application/ld+json' -i 
HTTP/2 200 
[...]
link: </docs/jsonldcontext.jsonld>; rel="alternate"; type="application/ld+json"
```

**Comment by rlzijdeman** [comment]
Thank you @progval. I have updated my comment accordingly.

**Comment by lubianat** [comment]
I am interested on the progress here, has there been any?

**Comment by leobard** [comment]
This morning I was happily programming  (🎩) , when a unit tests failed saying `expected https://schema.org, actual http://schema.org` . 

I thought: "*but I made a solid decision copying [solid-namespace](https://github.com/solid/solid-namespace/blob/main/index.js) !* ".  Reading their file again, it dawned on me: 
```
schema: 'http://schema.org/', // @@ beware confusion with documents no 303
```
 
Knowing that [URIS are cool if they 303](https://www.w3.org/TR/cooluris/) , I discovered a rabbit hole in the schema.org doc with helpful hyperlinks guiding me here. 

Good to see the discussion.

- I embrace the zen leadership of schema org writing in the [FAQ](https://schema.org/docs/faq.html#19) "*This is a lengthy way of saying that both 'https://schema.org' and 'http://schema.org' are fine*." . 
- I used to love URIs that don't change, but we live in a world of constant change. I happily accept *radical inclusion* of both `http`and `https` in the spirit of diversity. ❤️‍🔥 🏳‍🌈
- I understood "*Although we encourage a move towards 'https'*" on [developers](https://schema.org/docs/developers.html)

I thought about it and came up with this solution to fix that failing unit test of mine: 
- `https` is the future
- When reading and importing data (including [solid-namespace](https://github.com/solid/solid-namespace/blob/main/index.js) ), I will read both and transform `http://schema.org/` to `https://schema.org/`. 
- When writing, it is `https://schema.org/`. 
- I  will use  `https` in the majority of my code and data (tests, sparql queries, ...).
- If we all do it like this, we will eventually wake up in a world when all data is `https` and our linked data parsers can ditch the `http` for schema.org.

Is this clever?

If yes, to help others come to similar conclusions, would you consider changing the docs? Possible hints that would have helped me (and maybe the folks from [solid-namespace](https://github.com/solid/solid-namespace/blob/main/index.js)  ) :
- We recommend to publish and write `https://schema.org/` (not `http://schema.org/`).
- We recommend to transform from `http://schema.org/` to `https://schema.org/` on read, when importing data into a system.
- We from schema.org primarily publish specs using `https://schema.org/` now to support everyone during the transition. Also, all our namespaces are now `https://schema.org`, in all html docs and also in the main serializations available for download.
- We will observe how and when the community adapts to `https://schema.org/` and will time the deprecation of the `http://schema.org/` Namespace accordingly.


(🎩) ... I was happily programming a linkeddata + markdown personal note-taking system ... again. Even after previous adventures in semantic wikis and semantic desktops of varying outcome, I still ❤ using RDF for personal information management.
